src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/validators/PathValidator.java

```diff
@@ -0,0 +1,456 @@
+                data.getSegment().getSrcSwitchId(), data.getSegment().getSrcPort(),
+                data.getSegment().getDestSwitchId(), data.getSegment().getDestPort(),
+                data.getPath().getLatency().toMillis(), Duration.ofNanos(actualLatency).toMillis());
+    }
+
+    private String getForwardBandwidthErrorMessage(InputData data, long actualBandwidth) {
+        return String.format(
+                "There is not enough bandwidth between end points: switch %s port %d and switch %s port %d"
+                        + " (forward path). Requested bandwidth %d, but the link supports %d",
+                data.getSegment().getSrcSwitchId(), data.getSegment().getSrcPort(),
+                data.getSegment().getDestSwitchId(), data.getSegment().getDestPort(),
+                data.getPath().getBandwidth(), actualBandwidth);
+    }
+
+    private String getReverseBandwidthErrorMessage(InputData data, long actualBandwidth) {
+        return String.format(
+                "There is not enough bandwidth between end points: switch %s port %d and switch %s port %d"
+                        + " (reverse path). Requested bandwidth %d, but the link supports %d",
+                data.getSegment().getDestSwitchId(), data.getSegment().getDestPort(),
+                data.getSegment().getSrcSwitchId(), data.getSegment().getSrcPort(),
+                data.getPath().getBandwidth(), actualBandwidth);
+    }
+
+    private String getNoForwardIslError(InputData data) {
+        return String.format(
+                "There is no ISL between end points: switch %s port %d and switch %s port %d",
+                data.getSegment().getSrcSwitchId(), data.getSegment().getSrcPort(),
+                data.getSegment().getDestSwitchId(), data.getSegment().getDestPort());
+    }
+
+    private String getNoReverseIslError(InputData data) {
+        return String.format(
+                "There is no ISL between end points: switch %s port %d and switch %s port %d",
+                data.getSegment().getDestSwitchId(), data.getSegment().getDestPort(),
+                data.getSegment().getSrcSwitchId(), data.getSegment().getSrcPort());
+    }
+
+    private String getForwardIslNotActiveError(InputData data) {
+        return String.format(
+                "The ISL is not in ACTIVE state between end points: switch %s port %d and switch %s port %d",
+                data.getSegment().getSrcSwitchId(), data.getSegment().getSrcPort(),
+                data.getSegment().getDestSwitchId(), data.getSegment().getDestPort());
+    }
+
+    private String getReverseIslNotActiveError(InputData data) {
+        return String.format(
+                "The ISL is not in ACTIVE state between end points: switch %s port %d and switch %s port %d",
+                data.getSegment().getDestSwitchId(), data.getSegment().getDestPort(),
+                data.getSegment().getSrcSwitchId(), data.getSegment().getSrcPort());
+    }
```

Comment:
These functions are not being tested

Reply:
The code has changed and some methods are moved. But I added new test cases for the gaps in tests you mentioned.